api/src/main/java/com/cloud/network/NetworkModel.java

```diff
@@ -114,6 +114,13 @@ public interface NetworkModel {
 
     List<? extends Nic> getNics(long vmId);
 
+    /**
+     * gets the next available MAC and checks it for global uniqueness in the nics table. It will keep looking till it fins a nic to use.
```

Comment:
```suggestion
     * Gets the next available MAC and checks it for global uniqueness in the nics table. It will keep looking until it finds a MAC address that it is unique.
```

Reply:
```suggestion
     * Gets the next available MAC and checks it for global uniqueness in the nics table. It will keep looking until it finds a MAC address that is unique.
```
thanks for the improvement @BryanMLima , but i found a small quirk in the form of an extra it ;)